mmf/models/krisp.py

```diff
@@ -0,0 +1,764 @@
+# Copyright (c) Facebook, Inc. and its affiliates.
+
+import logging
+import math
+import os
+
+import torch
+import torch.nn as nn
+from mmf.common.registry import registry
+from mmf.models import BaseModel
+from mmf.modules.embeddings import BertVisioLinguisticEmbeddings
+from mmf.utils.configuration import get_mmf_cache_dir
+from mmf.utils.file_io import PathManager
+from mmf.utils.modeling import get_optimizer_parameters_for_bert
+from mmf.utils.transform import (
+    transform_to_batch_sequence,
+    transform_to_batch_sequence_dim,
+)
+from omegaconf import OmegaConf
+from transformers.modeling_bert import (
+    BertConfig,
+    BertEncoder,
+    BertForPreTraining,
+    BertLayer,
+    BertModel,
+    BertPooler,
+    BertPredictionHeadTransform,
+    BertPreTrainedModel,
+)
+
+
+# Import graph network module
+# Putting in try-catch to avoid adding dependencies to mmf
+try:
+    from projects.krisp.graphnetwork_module import GraphNetworkModule
```

Comment:
Ideally, you can move your model as well to projects and use user dir to import your model. This would be better approach. @vedanuj what should be better approach do you think is here? It is also not clear in which cases `projects` folder will be registered to python. You may want to `sys.append` manually here so that it is discoverable. Try some other things as well:

- Try pip packaging MMF and installing it to see if you can actually run this.

Reply:
Yes that was my recommendation as well in my previous comments. This model will anyway require additional installation and dependencies so we should move the model under projects folder and use the user directory approach.

Reply:
I did check and this relative import does work with pip installing. My FB part-time so I'm having to re-set everything up on an external machine, but will give us a sanity check that it works outside facebook.

I'll comment on the remaining comments once I've actually made and pushed those changes. Sorry it's taking me a while, got caught up with other things this week